sdks/python/apache_beam/io/iobase.py

```diff
@@ -659,6 +675,167 @@ def fraction_consumed(self):
     """
     raise NotImplementedError
 
+  def split_points(self):
+    """Gives the number of split points consumed and remaining.
+
+    For a ``RangeTracker`` used by a ``BoundedSource`` (within a
+    ``BoundedSource.read()`` invocation) this method produces a 2-tuple that
+    gives the number of split points consumed by the ``BoundedSource`` and the
+    number of split points remaining within the range of the ``RangeTracker``
+    that has not been consumed by the ``BoundedSource``.
+
+    More specifically, given that the position of the current record being read
+    by ``BoundedSource`` is current_position this method produces a tuple that
+    consists of
+    (1) number of split points in the range [self.start_position(),
+    current_position) without including the split point that is currently being
+    consumed. This represents the total amount of parallelism in the consumed
+    part of the source.
+    (2) number of split points within the range
+    [current_position, self.stop_position()) including the split point that is
+    currently being consumed. This represents the total amount of parallelism in
+    the unconsumed part of the source.
+
+    Similar to other methods of class ``RangeTracker``, this method may get
+    invoked by multiple threads hence must be thread safe, e.g. by using a
+    single lock object.
+
+    ** Determining consumed and remaining number of split points **
+
+    Consider the following examples: (1) An input that can be read in parallel
+    down to the individual records, is called "perfectly splittable".
+    (2) a "block-compressed" file format such as ``avroio``, in which a block of
+    records has to be read as a whole, but different blocks can be read in
+    parallel. (3) an "unsplittable" input such as a cursor in a database.
+
+    consumed number of split points:
```

Comment:
Capitalize? Or are you trying to enumerate some examples here, in which case I would rephrase this.